crates/js-sys/src/lib.rs

```diff
@@ -3649,6 +3674,12 @@ impl PartialEq<str> for JsString {
     }
 }
 
+impl PartialEq<char> for JsString {
```

Comment:
This feels pretty awkward. It's similar to implementing `PartialEq<u32> for Vec<u32>` (which would obviously be wrong).

`String` and `&str` in Rust do not define `PartialEq` for `char`, so this is unprecedented.

On the other hand, I can see the practical use for it, so I'm undecided on whether it's a good idea or not.

Reply:
> String and &str in Rust do not define PartialEq for char, so this is unprecedented.

I think this is a bit different, because, unlike with str/char, with JsString we have a single type representation for both, so it seems to make sense to define traits for both too.

Reply:
In JS, sure, but this is comparing a Rust `char` to a JS string, so they're actually not a single type. It's a tricky situation.

Reply:
Yeah, but where is `From`, it seems right to have `PartialEq` as well. 

Idk, if you feel strongly opposed, I can remove it, but it seems useful to have these symmetrical.